common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java

```diff
@@ -236,10 +248,16 @@ public static void runTests(PerfTestBase<?>[] tests, boolean sync, int parallel,
         try {
             if (sync) {
                 ForkJoinPool forkJoinPool = new ForkJoinPool(parallel);
-                forkJoinPool.submit(() -> {
-                    IntStream.range(0, parallel).parallel().forEach(i -> tests[i].runAll(endNanoTime));
-                }).get();
+                List<Callable<Integer>> operations = new ArrayList<>(parallel);
```

Comment:
Why was this changed?

Reply:
This was changed as the previous implementation didn't behave as expected. What it did before was create a single task that then ran whatever limit `.parallel()` had for parallelization. It now runs `parallel` tasks concurrently as expected.

Reply:
Are you saying the old code didn't actually run the tasks in parallel?  Or was it limited to a parallelism that might have been lower than requested?  Because I though I verified the sync tests ran in parallel as expected with the old code, but we mostly run async tests so this might have been a bug.

Reply:
The latter, it ran in parallel, but the parallelism level wasn't in our control but in the control of however `Stream.parallel` is handled by the JDK.